metadata/src/test/java/edu/unc/lib/dl/util/URIUtilTest.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright 2016 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.util;
+
+import static org.junit.Assert.assertEquals;
+
+import java.net.URI;
+
+import org.junit.Test;
+
+public class URIUtilTest {
+
+	@Test
+	public void joinTest() {
+		assertEquals("http://example.com/base/path",
+				URIUtil.join("http://example.com/base", "path"));
+
+		assertEquals("http://example.com/base/path/to",
+				URIUtil.join("http://example.com/", "base/", "/path", "to/"));
+
+		assertEquals("path/to", URIUtil.join("", "path", "to"));
+
+		assertEquals("http://example.com/base/path",
+				URIUtil.join(URI.create("http://example.com/base"), "path"));
+	}
```

Comment:
My one suggestion would be to cover a few more test secnarios:
(1)  a null base String
(2)  an empty base String
(3) a null array of segments
(4) an empty array of segments